transformer_engine/common/common.h

```diff
@@ -281,6 +282,132 @@ struct Tensor {
   }
 };
 
+struct GroupedTensor {
+ public:
+  /*
+  Grouped tensor is a collection of tensors with different shapes but the same dtype and scaling mode
+
+  Shape Representation:
+  - logical_shape: 2D shape representing the conceptual layout (REQUIRED)
+    + When all_same_shape(): [num_tensors * M, N] where each tensor is (M, N)
+    + When varying_first_dim(): [~sum_of_first_dims, N] where N is common
+    + When varying_last_dim(): [M, ~sum_of_last_dims] where M is common
+    + When varying_both_dims(): [1, total_elements] (fully flattened)
```

Comment:
I don't think "logical shape" is quite the right name for this concept. When I hear "logical shape", I think of the mathematical tensor that we are computing, which is completely independent of the physical data layout. An activation tensor might have logical shape `[batch_size, seqlen, num_heads, hidden_dim]`, even if the wgrad GEMM is actually using an FP8 transpose with shape `[hidden_dim, batch_size*seqlen*num_heads]`.

Really, what we're saying is that we have a full tensor with shape `[m, n]`, and we are slicing this tensor. So something like `full_tensor_logical_shape` might be the most accurate name, although it's too cumbersome. How about `full_tensor_shape`?

Reply:
Ok, maybe we can just make the comments a little clearer:
```suggestion
  - logical_shape: 2D shape representing the conceptual layout, i.e. the shape when the grouped tensors are flattened to 2D and concatenated together (REQUIRED)
    + When all_same_shape(): [num_tensors * M, N] where each tensor is (M, N)
    + When varying_first_dim(): [~sum_of_first_dims, N] where N is common
    + When varying_last_dim(): [M, ~sum_of_last_dims] where M is common
    + When varying_both_dims(): [1, total_elements] (fully flattened)
```